src/dodal/beamlines/i22.py

```diff
@@ -24,26 +28,65 @@
 from dodal.utils import BeamlinePrefix, get_beamline_name, skip_device
 
 BL = get_beamline_name("i22")
+print("BL NAME: ", BL)
 set_log_beamline(BL)
 set_utils_beamline(BL)
 
+_LAB_NAME = "p38"
+
+IS_LAB = BL == _LAB_NAME
+print("is this lab? : ", IS_LAB)
+
+LINKAM_IS_IN_LAB = False
 # Currently we must hard-code the visit, determining the visit at runtime requires
 # infrastructure that is still WIP.
 # Communication with GDA is also WIP so for now we determine an arbitrary scan number
 # locally and write the commissioning directory. The scan number is not guaranteed to
 # be unique and the data is at risk - this configuration is for testing only.
-set_directory_provider(
+directory_provider = (
     StaticVisitDirectoryProvider(
+        BL,
+        Path("/dls/p38/data/2024/cm37282-2/bluesky"),
+        client=DirectoryServiceClient("https://p38-control:8088/api"),
+    )
+    if IS_LAB
+    else StaticVisitDirectoryProvider(
         BL,
         Path("/dls/i22/data/2024/cm37271-2/bluesky"),
         client=DirectoryServiceClient("http://i22-control:8088/api"),
     )
 )
 
+set_directory_provider(directory_provider)
 
+
+# d11 at p38, but disconnected
+@skip_device(lambda: BL == _LAB_NAME)
 def saxs(
     wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> PilatusDetector:
+) -> NXSasAravis | NXSasPilatus:
+    if IS_LAB:
+        return device_instantiation(
+            NXSasAravis,
+            "saxs",
+            "-DI-DCAM-03:",
+            wait_for_connection,
+            fake_with_ophyd_sim,
+            drv_suffix="DET:",
+            hdf_suffix="HDF5:",
+            # todo double check the metadata
```

Comment:
Should be able to get most of these figures from the i22 oav which is a similar model. Then can remove the TODO, or otherwise make a ticket to track the TODO and link it here.